src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc

```diff
@@ -0,0 +1,209 @@
+
+namespace sdl_rpc_plugin {
+
+SDL_CREATE_LOG_VARIABLE("SdlRPCPlugin")
+
+SDLPendingResumptionHandler::SDLPendingResumptionHandler(
+    application_manager::ApplicationManager& application_manager)
+    : PendingResumptionHandler(application_manager) {}
+
+smart_objects::SmartObjectSPtr
+SDLPendingResumptionHandler::CreateSubscriptionRequest() {
+  SDL_LOG_AUTO_TRACE();
+  auto subscribe_waypoints_msg =
+      application_manager::MessageHelper::CreateMessageForHMI(
+          hmi_apis::FunctionID::Navigation_SubscribeWayPoints,
+          application_manager_.GetNextHMICorrelationID());
+  (*subscribe_waypoints_msg)[application_manager::strings::params]
+                            [application_manager::strings::message_type] =
+                                hmi_apis::messageType::request;
+  return subscribe_waypoints_msg;
+}
+
+void SDLPendingResumptionHandler::HandleResumptionSubscriptionRequest(
+    application_manager::AppExtension& extension,
+    application_manager::Application& app) {
+  SDL_LOG_AUTO_TRACE();
+  WayPointsAppExtension& ext = dynamic_cast<WayPointsAppExtension&>(extension);
+  smart_objects::SmartObjectSPtr request = CreateSubscriptionRequest();
+  smart_objects::SmartObject& request_ref = *request;
+  const auto function_id = static_cast<hmi_apis::FunctionID::eType>(
+      request_ref[application_manager::strings::params]
+                 [application_manager::strings::function_id]
+                     .asInt());
+  const uint32_t corr_id =
+      request_ref[application_manager::strings::params]
+                 [application_manager::strings::correlation_id]
+                     .asUInt();
+
+  auto resumption_request =
+      MakeResumptionRequest(corr_id, function_id, *request);
+  app_ids_.push(app.app_id());
+
+  if (pending_requests_.empty()) {
+    SDL_LOG_DEBUG("There are no pending requests for app_id: " << app.app_id());
+    pending_requests_[corr_id] = request_ref;
+    subscribe_on_event(function_id, corr_id);
+    SDL_LOG_DEBUG("Sending request with function id: "
+                  << function_id << " and correlation_id: " << corr_id);
+
+    application_manager_.GetRPCService().ManageHMICommand(request);
+  } else {
+    SDL_LOG_DEBUG("There are pending requests. Freeze resumption for app id "
+                  << app.app_id() << " corr id = " << corr_id);
+    ResumptionAwaitingHandling frozen_res{
+        app.app_id(), ext, resumption_request};
+    frozen_resumptions_.push_back(frozen_res);
+  }
+  resumption_data_processor().SubscribeToResponse(app.app_id(),
+                                                  resumption_request);
+}
+
+void SDLPendingResumptionHandler::OnResumptionRevert() {
+  SDL_LOG_AUTO_TRACE();
+  using namespace application_manager;
+
+  if (!pending_requests_.empty()) {
+    SDL_LOG_DEBUG("Still waiting for some response");
+    return;
+  }
+
+  if (!frozen_resumptions_.empty()) {
+    ResumptionAwaitingHandling freezed_resumption = frozen_resumptions_.back();
```

Comment:
```suggestion
    ResumptionAwaitingHandling frozen_resumption = frozen_resumptions_.back();
```

Reply:
@iCollin renamed in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/8913095240e15b52211f29fb6af06a4bd573155a